packages/manager/apps/dedicated-servers/src/components/dataGridColumns.tsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import { FilterTypeCategories } from '@ovh-ux/manager-core-api';
+import { OdsBadge } from '@ovhcloud/ods-components/react';
+import { ODS_BADGE_COLOR } from '@ovhcloud/ods-components';
+import {
+  DataGridTextCell,
+  DatagridColumn,
+} from '@ovh-ux/manager-react-components';
+import { ActionCell } from '@/components/actionCell';
+import { DedicatedServer } from '@/data/types/server.type';
+import MonitoringStatusChip from '@/components/monitoringStatus';
+import { DSVrack } from './vRackCell';
+import NameCell from './cells/nameCell';
+import RenewCell from './cells/renewCell';
+import ExpirationCell from './cells/expirationCell';
+import EngagementCell from './cells/engagementCell';
+import PriceCell from './cells/priceCell';
+
+const colorByProductStatus: Record<string, ODS_BADGE_COLOR> = {
+  ok: ODS_BADGE_COLOR.success,
+  hacked: ODS_BADGE_COLOR.warning,
+  hackedBlocked: ODS_BADGE_COLOR.information,
+};
+
+const textByProductStatus: Record<string, string> = {
+  ok: 'server_configuration_state_OK',
+  hacked: 'server_configuration_state_HACKED',
+  hackedBlocked: 'server_configuration_state_HACKED_BLOCKED',
+};
+
+export function getColumns(
+  t: (v: string) => string,
+  goToServer: (name: string) => void,
+) {
+  const serverColumns: DatagridColumn<DedicatedServer>[] = [
+    {
+      id: 'serverId',
+      isSearchable: true,
+      isFilterable: true,
+      enableHiding: true,
+      type: FilterTypeCategories.Numeric,
+      label: t('server_display_id'),
+      cell: (server: DedicatedServer) => (
+        <DataGridTextCell>{t(server.serverId.toString())}</DataGridTextCell>
+      ),
+    },
+    {
+      id: 'iam.displayName',
+      isSearchable: true,
+      isFilterable: true,
+      enableHiding: true,
+      type: FilterTypeCategories.String,
+      label: t('server_display_name'),
+      cell: (server: DedicatedServer) => (
+        <NameCell server={server} navigate={goToServer} />
+      ),
```

Comment:
```suggestion
      cell: NameCell,
```

Don't pass `goToServer` function and just take server as props `function NameCell(server: DedicatedServer) => jsx` or `const NameCell: React.FC<DedicatedServer> = (server) => jsx`

The shell is in the context to be accessible to all components so you don't need to pass `goToServer` as a prop

If you would like to factor the code you can create a hook `useGoToServer(server)` that will return the function to navigate to the right dashboard